tensorflow_federated/python/core/impl/executors/federating_executor.py

```diff
@@ -523,273 +842,70 @@ def _check_value_compatible_with_placement(self, value, placement, all_equal):
-
-    async def _move(v):
-      return await child.create_value(await v.compute(), item_type)
-
-    items = await asyncio.gather(*[_move(v) for v in val])
-
-    zero = await child.create_value(
-        await (await self.create_selection(arg, index=1)).compute(), zero_type)
-    op = await child.create_value(arg.internal_representation[2], op_type)
-
-    result = zero
-    for item in items:
-      result = await child.create_call(
-          op, await child.create_tuple(
-              anonymous_tuple.AnonymousTuple([(None, result), (None, item)])))
-    return FederatingExecutorValue([result],
-                                   computation_types.FederatedType(
-                                       result.type_signature,
-                                       placement_literals.SERVER,
-                                       all_equal=True))
+    return await self.intrinsic_strategy.federated_reduce(arg)
 
   @tracing.trace
-  async def _compute_intrinsic_federated_secure_sum(self, arg):
-    raise NotImplementedError('The secure sum intrinsic is not implemented.')
+  async def _compute_intrinsic_federated_aggregate(self, arg):
+    return await self.intrinsic_strategy.federated_aggregate(arg)
 
   @tracing.trace
   async def _compute_intrinsic_federated_sum(self, arg):
-    py_typecheck.check_type(arg.type_signature, computation_types.FederatedType)
-    zero, plus = tuple(await
-                       asyncio.gather(*[
-                           executor_utils.embed_tf_scalar_constant(
-                               self, arg.type_signature.member, 0),
-                           executor_utils.embed_tf_binary_operator(
-                               self, arg.type_signature.member, tf.add)
-                       ]))
-    return await self._compute_intrinsic_federated_reduce(
-        FederatingExecutorValue(
-            anonymous_tuple.AnonymousTuple([
-                (None, arg.internal_representation),
-                (None, zero.internal_representation),
-                (None, plus.internal_representation)
-            ]),
-            computation_types.NamedTupleType(
-                (arg.type_signature, zero.type_signature, plus.type_signature)))
-    )
+    return await self.intrinsic_strategy.federated_sum(arg)
 
   @tracing.trace
-  async def _compute_intrinsic_federated_value_at_clients(self, arg):
-    return await executor_utils.compute_intrinsic_federated_value(
-        self, arg, placement_literals.CLIENTS)
-
-  @tracing.trace
-  async def _compute_intrinsic_federated_value_at_server(self, arg):
-    return await executor_utils.compute_intrinsic_federated_value(
-        self, arg, placement_literals.SERVER)
+  async def _compute_intrinsic_federated_mean(self, arg):
+    return await self.intrinsic_strategy.federated_mean(arg)
 
   @tracing.trace
   async def _compute_intrinsic_federated_weighted_mean(self, arg):
     return await executor_utils.compute_intrinsic_federated_weighted_mean(
```

Comment:
This should be delegated to intrinsic_strategy and implemented on the strategy to delegate to this function. In the long term I think we can probably inline the implementation of this function in the strategy.

Reply:
I think this was an oversight -- there is already a `federated_weighted_mean` implemented on the strategy, we just never properly delegated